add discussion in interpretability section and updates to molecular design section and discussion sections (issues with previous PR fixed) (#988)
greenelab/deep-review: content/05.treat.md

The Grammar Variational Autoencoder, which takes the SMILES grammar into account and is guaranteed to produce syntactically valid SMILES, helps alleviate this issue to some extent [@arxiv:1703.01925].

Another approach to *de novo* design is to train character-based RNNs on large collections of molecules, for example, ChEMBL [@doi:10.1093/nar/gkr777], to first obtain a generic generative model for drug-like compounds [@tag:Segler2017_drug_design].
These generative models successfully learn the grammar of compound representations, with 94% [@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of generated SMILES corresponding to valid molecular structures. The initial RNN is then fine-tuned to generate molecules that are likely to be active against a specific target by either continuing training on a small set of positive examples [@tag:Segler2017_drug_design] or adopting reinforcement learning strategies [@tag:Olivecrona2017_drug_design; @arxiv:1611.02796]. Both the fine-tuning and reinforcement learning approaches can rediscover known, held-out active molecules.

> 

Reviewer: >Both the **fine-tuning and reinforcement learning** approaches can rediscover known, held-out active molecules.

Nitpicky but for me there isn't a need to distinguish here since they're both fine-tuning approaches. It may lead to downstream confusion between RL as a fine tuning procedure (e.g. using policy gradients with rewards from an auxiliary predictor) vs. ab initio RL approaches with no pre-training like MolDQN.

Author: Yes, I agree. I think this was old text. The newer RL approaches are a different breed - from scratch. Will you try to fix it?

> Reinforcement learning approaches where operations are performed directly on the molecular graph bypass the need to learn the details of SMILES syntax, allowing the model to focus purely on chemistry. Additionally, they seem to require less training data and generate more valid molecules since they are constrained by design only to graph operations which satisfy chemical valiance rules.[@tag:Elton_molecular_design_review] A reinforcement learning agent developed by Zhou et al. demonstrated superior molecular optimization performance on certain easy to compute metrics when compared with other deep learning based approaches such as the Junction Tree VAE, Objective Reinforced Generative Adversarial Network, and Graph Convolutional Policy Network.[@doi:10.1038/s41598-019-47148-x] As another example, Zhavoronkov et al. used generative tensorial reinforcement learning to discover potent inhibitors of discoidin domain receptor 1 (DDR1).[@tag:Zhavoronkov2019_drugs] Their work is unique in that six lead candidates discovered using their approach were synthesized and tested in the lab, with 4/6 achieving some degree of binding to DDR1.[@tag:Zhavoronkov2019_drugs]  

Reviewer: >A reinforcement learning agent developed by Zhou et al. demonstrated superior molecular optimization performance on **certain easy to compute** metrics

I think it'd be better to directly specify the metrics (e.g. logP, QED) here.

Author: sure, the metrics they tested were QED and "penalized logP" (logP minus the synthetic accessibility). They are not the most useful but easy to compute and enough to make the comparison they wanted.

Reviewer 2: The QED acronym doesn't appear in the manuscript, so let's call this "quantitative estimation of drug-likeness".  We may also want to explain or define logP in a brief phrase.

#### *De novo* drug design

> *De novo* drug design attempts to model the typical design-synthesize-test cycle of drug discovery in-silico [@doi:10.1002/wcms.49; @doi:10.1021/acs.jmedchem.5b01849].

Reviewer 2: ```suggestion
*De novo* drug design attempts to model the typical design-synthesize-test cycle of drug discovery *in silico* [@doi:10.1002/wcms.49; @doi:10.1021/acs.jmedchem.5b01849].
```
Formatting consistency with other parts of the manuscript.

It explores an estimated 10<sup>60</sup> synthesizable organic molecules with drug-like properties without explicit enumeration [@doi:10.1002/wcms.1104].

> To score molecules after generation or during optimization, physics-based simulation could be used [@tag:Sumita2018], but machine learning models based on techniques discussed earlier may be preferable [@tag:Gomezb2016_automatic], as they are much more computationally expedient. Computationally efficiency is particularly important during optimization as the "scoring function" may need to be called thousands of times.

Reviewer 2: ```suggestion
To score molecules after generation or during optimization, physics-based simulation could be used [@tag:Sumita2018], but machine learning models based on techniques discussed earlier may be preferable [@tag:Gomezb2016_automatic], as they are much more computationally expedient.
Computational efficiency is particularly important during optimization as the "scoring function" may need to be called thousands of times.
```
Add newline and rephrase

To "design" and "synthesize", traditional *de novo* design software relied on classical optimizers such as genetic algorithms.

> These approaches can lead to overfit, "weird" molecules, which are difficult to synthesize in the lab.

Reviewer 2: Does this partly contradict the new text on genetic algorithms below?  What is our overall stance on genetic algorithms versus generative neural networks?

Author: That sentence I think was actually from the earlier version or it was accidently put here by mistake. It's funny because first autoencoder for molecular generation got the exact same criticism. In the papers on the ORGAN/ORGANIC GAN model they showed its' very easy to get "weird"/non-physical/impossible-to-synthesize molecules which are "over-optimized" for a certain thing if you don't put constraints in the loss function to prevent that.  Anyway, I would remove the sentence about 'weird' molecules.

Reviewer 2: I agree, you can remove the sentence about "weird" molecules in your next round of edits.  Or alternatively note that this is still a problem with many NN-based methods.

> A popular approach which may help ensure synthesizability is to use rule-based virtual chemical reactions to generate molecular structures [@doi:10.1021/acs.jmedchem.5b01849].

Reviewer 2: ```suggestion
A popular approach that may help ensure synthesizability is to use rule-based virtual chemical reactions to generate molecular structures [@doi:10.1021/acs.jmedchem.5b01849].
```

Deep learning models that generate realistic, synthesizable molecules have been proposed as an alternative.
In contrast to the classical, symbolic approaches, generative models learned from data would not depend on laboriously encoded expert knowledge.

> In the past few years a large number of techniques for the generative modeling and optimization of molecules with deep learning have been explored, including recursive neural networks, variational autoencoders, generative adversarial networks, and reinforcement learning -- for a review see Elton, et al.[@tag:Elton_molecular_design_review] or Vamathevan et al.[@tag:Vamathevan2019].

Reviewer 2: ```suggestion
In the past few years, a large number of techniques for the generative modeling and optimization of molecules with deep learning have been explored, including recursive neural networks, VAEs, GANs, and reinforcement learning -- for a review see Elton et al. [@tag:Elton_molecular_design_review] or Vamathevan et al [@tag:Vamathevan2019].
```
Confirming you mean recursive and not recurrent neural networks.  I'm not sure what generative model that refers to.

Author: Sorry, I meant _recurrent_ neural networks.

Reviewer 2: I updated my suggestion to use "RNNs"

> 

Reviewer 2: Can we remove the blank line here and after the next sentence?

Author: yeah, sure.

> Building off the large amount of work that has already gone into text generation,[@arxiv:1308.0850] many generative neural networks for drug design represent chemicals with the simplified molecular-input line-entry system (SMILES), a standard string-based representation with characters that represent atoms, bonds, and rings [@tag:Segler2017_drug_design].

Reviewer 2: ```suggestion
Building off the large amount of work that has already gone into text generation [@arxiv:1308.0850], many generative neural networks for drug design initially represented chemicals with the simplified molecular-input line-entry system (SMILES), a standard string-based representation with characters that represent atoms, bonds, and rings [@tag:Segler2017_drug_design].
```
My understanding is that the field has somewhat moved past generating SMILES and most of the attention is now on graphs.  Do you agree?

Author: Yes, absolutely.

Another approach to *de novo* design is to train character-based RNNs on large collections of molecules, for example, ChEMBL [@doi:10.1093/nar/gkr777], to first obtain a generic generative model for drug-like compounds [@tag:Segler2017_drug_design].
These generative models successfully learn the grammar of compound representations, with 94% [@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of generated SMILES corresponding to valid molecular structures.
The initial RNN is then fine-tuned to generate molecules that are likely to be active against a specific target by either continuing training on a small set of positive examples [@tag:Segler2017_drug_design] or adopting reinforcement learning strategies [@tag:Olivecrona2017_drug_design; @arxiv:1611.02796].
Both the fine-tuning and reinforcement learning approaches can rediscover known, held-out active molecules.

Reinforcement learning approaches where operations are performed directly on the molecular graph bypass the need to learn the details of SMILES syntax, allowing the model to focus purely on chemistry.

> Additionally, they seem to require less training data and generate more valid molecules since they are constrained by design only to graph operations which satisfy chemical valiance rules.[@tag:Elton_molecular_design_review]

Reviewer 2: ```suggestion
Additionally, they seem to require less training data and generate more valid molecules since they are constrained by design only to graph operations which satisfy chemical valiance rules [@tag:Elton_molecular_design_review].
```

> A reinforcement learning agent developed by Zhou et al. demonstrated superior molecular optimization performance on certain easy to compute metrics when compared with other deep learning based approaches such as the Junction Tree VAE, Objective Reinforced Generative Adversarial Network, and Graph Convolutional Policy Network [@doi:10.1038/s41598-019-47148-x].

Reviewer 2: ```suggestion
A reinforcement learning agent developed by Zhou et al. [@doi:10.1038/s41598-019-47148-x] demonstrated superior molecular optimization performance on certain easy to compute metrics when compared with other deep learning based approaches such as the Junction Tree VAE [add ref], Objective Reinforced Generative Adversarial Network [add ref], and Graph Convolutional Policy Network [add ref].
```
Can you please add the three references for this related work here?  Some of those models are/were also fairly important in the field and are worthwhile citations.

Reviewer: >Objective Reinforced Generative Adversarial Network
Objective-Reinforced

Reviewer 2: I updated my suggestion to include this

As another example, Zhavoronkov et al. used generative tensorial reinforcement learning to discover potent inhibitors of discoidin domain receptor 1 (DDR1) [@tag:Zhavoronkov2019_drugs].

> Their work is unique in that six lead candidates discovered using their approach were synthesized and tested in the lab, with 4/6 achieving some degree of binding to DDR1 [@tag:Zhavoronkov2019_drugs].

Reviewer 2: Referring back to https://github.com/greenelab/deep-review/pull/985#discussion_r379683813, perhaps we don't need to get into the details of the Walters and Murcko response.  However, I do think it could be valuable to mention the challenges in evaluating generative models in this domain.  Maybe part of the concluding paragraph should be about those challenges.  It could reference your review for metrics that are used and the idea to report novelty with respect to the training data, referencing Walters and Murcko.

> In concluding this section, it is worth pointing out that it has been shown that classical genetic algorithms can compete with some of the most advanced deep learning methods for molecular optimization [@doi:10.1246/cl.180665; @doi:10.1039/C8SC05372C].

Reviewer 2: I'm not familiar with these methods.  Do their evaluations seem rigorous to you?  The first example compared to three SMILES-based methods, which aren't exactly state-of-the-art at the time of this writing.

We can also shorten the first part of the sentence.

Such genetic algorithms use hard coded rules based possible chemical reactions to generate molecular structures [@doi:10.1021/acs.jmedchem.5b01849].

> Still, there are many avenues for improving current deep learning systems and the future of the field looks bright.

Reviewer 2: Can we conclude with a more specific outlook?  We don't talk much about the transition to graph-based generative models, so that would be one idea.  Another would be the evaluation challenges I noted above.

It explores an estimated 10<sup>60</sup> synthesizable organic molecules with drug-like properties without explicit enumeration [@doi:10.1002/wcms.1104].
To score molecules after generation or during optimization, physics-based simulation could be used [@tag:Sumita2018], but machine learning models based on techniques discussed earlier may be preferable [@tag:Gomezb2016_automatic], as they are much more computationally expedient.
Computational efficiency is particularly important during optimization as the "scoring function" may need to be called thousands of times.

To "design" and "synthesize", traditional *de novo* design software relied on classical optimizers such as genetic algorithms.
These algorithms use a list of hard-coded rules to perform virtual chemical reactions on molecular structures during each iteration, leading to physically stable and synthesizable molecules [@doi:10.1021/acs.jmedchem.5b01849].
Deep learning models have been proposed as an alternative.
In contrast to the classical approaches, in theory generative models learned from big data would not require laboriously encoded expert knowledge to generate realistic, synthesizable molecules.

> In the past few years, a large number of techniques for the generative modeling and optimization of molecules with deep learning have been explored, including RNNs, VAEs, GANs, and reinforcement learning -- for a review see Elton et al. [@tag:Elton_molecular_design_review] or Vamathevan et al [@tag:Vamathevan2019].

Reviewer 2: ```suggestion
In the past few years, a large number of techniques for the generative modeling and optimization of molecules with deep learning have been explored, including RNNs, VAEs, GANs, and reinforcement learning---for a review see Elton et al. [@tag:Elton_molecular_design_review] or Vamathevan et al. [@tag:Vamathevan2019].
```

Building off the large amount of work that has already gone into text generation [@arxiv:1308.0850], many generative neural networks for drug design initially represented chemicals with the simplified molecular-input line-entry system (SMILES), a standard string-based representation with characters that represent atoms, bonds, and rings [@tag:Segler2017_drug_design].

> The first successful demonstration of a deep learning based approach for molecular optimization occurred in 2016 with the development of a SMILES-to-SMILES autoencoder capable of learning a continuous latent feature space for molecules[@tag:Gomezb2016_automatic].

Reviewer 2: ```suggestion
The first successful demonstration of a deep learning based approach for molecular optimization occurred in 2016 with the development of a SMILES-to-SMILES autoencoder capable of learning a continuous latent feature space for molecules [@tag:Gomezb2016_automatic].
```

These generative models successfully learn the grammar of compound representations, with 94% [@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of generated SMILES corresponding to valid molecular structures.
The initial RNN is then fine-tuned to generate molecules that are likely to be active against a specific target by either continuing training on a small set of positive examples [@tag:Segler2017_drug_design] or adopting reinforcement learning strategies [@tag:Olivecrona2017_drug_design; @arxiv:1611.02796].
Both the fine-tuning and reinforcement learning approaches can rediscover known, held-out active molecules.

Reinforcement learning approaches where operations are performed directly on the molecular graph bypass the need to learn the details of SMILES syntax, allowing the model to focus purely on chemistry.
Additionally, they seem to require less training data and generate more valid molecules since they are constrained by design only to graph operations which satisfy chemical valiance rules [@tag:Elton_molecular_design_review].

> A reinforcement learning agent developed by Zhou et al. [@doi:10.1038/s41598-019-47148-x] demonstrated superior molecular optimization performance on optimizing  the quantitative estimate of drug-likeness (QED) metric and the "penalized logP" metric (logP minus the synthetic accessibility) when compared with other deep learning based approaches such as the Junction Tree VAE [@arxiv:1802.04364], Objective-Reinforced Generative Adversarial Network [@arxiv:1705.10843], and Graph Convolutional Policy Network [@arxiv:1806.02473].

Reviewer 2: ```suggestion
A reinforcement learning agent developed by Zhou et al. [@doi:10.1038/s41598-019-47148-x] demonstrated superior molecular optimization performance on optimizing the quantitative estimate of drug-likeness (QED) metric and the "penalized logP" metric (logP minus the synthetic accessibility) when compared with other deep learning based approaches such as the Junction Tree VAE [@arxiv:1802.04364], Objective-Reinforced Generative Adversarial Network [@arxiv:1705.10843], and Graph Convolutional Policy Network [@arxiv:1806.02473].
```

As another example, Zhavoronkov et al. used generative tensorial reinforcement learning to discover inhibitors of discoidin domain receptor 1 (DDR1) [@tag:Zhavoronkov2019_drugs].

> In contrast to most previous work, six lead candidates discovered using their approach were synthesized and tested in the lab, with 4/6 achieving some degree of binding to DDR1. One of the molecules was chosen for further testing and showed promising results in a cancer cell line and mouse model [@tag:Zhavoronkov2019_drugs]. 

Reviewer 2: ```suggestion
In contrast to most previous work, six lead candidates discovered using their approach were synthesized and tested in the lab, with 4/6 achieving some degree of binding to DDR1.
One of the molecules was chosen for further testing and showed promising results in a cancer cell line and mouse model [@tag:Zhavoronkov2019_drugs]. 
```

> In concluding this section, we want to highlight two areas where work is still needed before AI can bring added value to the existing drug discovery process - novelty and synthesizability.

Reviewer 2: ```suggestion
In concluding this section, we want to highlight two areas where work is still needed before AI can bring added value to the existing drug discovery process---novelty and synthesizability.
```

> The work of Zhavoronkov et al. is a arguably an important milestone and recieved much fanfare in the popular press, but Walters and Murko have presented a more sober assessment, noting that the generated molecule they choose to test in the lab is very similar to an existing drug which was present in their training data [@doi:10.1038/s41587-020-0418-2].

Reviewer 2: ```suggestion
The work of Zhavoronkov et al. is arguably an important milestone and received much fanfare in the popular press, but Walters and Murko have presented a more sober assessment, noting that the generated molecule they choose to test in the lab is very similar to an existing drug that was present in their training data [@doi:10.1038/s41587-020-0418-2].
```

Small variations of existing molecules are likely not to be much better and may not be patentable.
One way to tackle this problem is to add novelty and diversity metrics to the reward function of reinforcement learning based algorithms.

> Novelty should also be taken into account when comparing different models - and thus is included in the proposed GuacaMol benchmark (2019) for accessing generative molecules for molecular design [@doi:10.1021/acs.jcim.8b00839].

Reviewer 2: ```suggestion
Novelty should also be taken into account when comparing different models---and thus is included in the proposed GuacaMol benchmark (2019) for accessing generative molecules for molecular design [@doi:10.1021/acs.jcim.8b00839].
```

The other area which has been pointed to as a key limitation of current approaches is synthesizability [@doi:10.1021/acs.jcim.0c00174; @doi:10.1021/acsmedchemlett.0c00088].

> Current heuristics of synthesizability, such as the synthetic accessibility score, are based on a relatively limited domain of chemical data and are too restrictive, so better models/heuristics of synthesizability should help in this area [doi:10.1021/acs.jcim.0c00174]. 

Reviewer 2: ```suggestion
Current heuristics of synthesizability, such as the synthetic accessibility score, are based on a relatively limited domain of chemical data and are too restrictive, so better modelsof synthesizability should help in this area [@doi:10.1021/acs.jcim.0c00174]. 
```

> As noted before, genetic algorithms use hard coded rules based on possible chemical reactions to generate molecular structures and therefore may have less trouble generating synthesizable molecules [@doi:10.1021/acs.jmedchem.5b01849].

Reviewer 2: ```suggestion
As noted before, genetic algorithms use hard-coded rules based on possible chemical reactions to generate molecular structures and therefore may have less trouble generating synthesizable molecules [@doi:10.1021/acs.jmedchem.5b01849].
```

> We note in passing that Jensen et al. (2018) [@doi:10.1039/C8SC05372C] and Yoshikawa et al. (2019) [@doi:10.1246/cl.180665] have both demostrated genetic algorithms which are competative with deep learning approaches. 

Reviewer 2: ```suggestion
We note in passing that Jensen (2018) [@doi:10.1039/C8SC05372C] and Yoshikawa et al. (2019) [@doi:10.1246/cl.180665] have both demonstrated genetic algorithms that are competitive with deep learning approaches. 
```

Removed et al for the single author paper
